tonic/src/transport/channel/endpoint.rs

```diff
@@ -51,13 +61,68 @@ impl Endpoint {
     {
         let me = dst.try_into().map_err(|e| Error::from_source(e.into()))?;
         #[cfg(feature = "_tls-any")]
-        if me.uri.scheme() == Some(&http::uri::Scheme::HTTPS) {
-            return me.tls_config(ClientTlsConfig::new().with_enabled_roots());
+        if let EndpointType::Uri(uri) = &me.uri {
+            if uri.scheme() == Some(&http::uri::Scheme::HTTPS) {
+                return me.tls_config(ClientTlsConfig::new().with_enabled_roots());
+            }
         }
-
         Ok(me)
     }
 
+    fn new_uri(uri: Uri) -> Self {
```

Comment:
The `new_uri` and `new_uds` constructors are a bit repetitive but I personally have no better idea on this.

Reply:
This feels fine for now.